src/sqlparser/src/ast/mod.rs

```diff
@@ -1666,22 +1666,61 @@ pub enum DescribeKind {
     Fragments,
 }
 
+task_local::task_local! {
+    /// A flag indicating whether to validate the unparsed SQL to be parsable
+    /// in [`Statement::fmt`] implementation. True by default.
+    static VALIDATE_PARSABLE: bool
```

Comment:
What's the benefit of using this task_local? Why not simply move all previous logic (except `expect`) to `try_to_string` and call it from `Display::fmt`?

Reply:
Oh it's for passing down extra info to recursive call without changing the signature 😂